File: src/action/EventTrigger.php
<?php


namespace Ething\Action;
	
class EventTrigger extends Action {
	
	
	// attributes validation function
	static public function validate(array &$attributes, array $context){
		
		$attributes = array_merge(
			array( 'name' => null ),
			$attributes
		);
		
		foreach(array_keys($attributes) as $key){
			
			switch($key){
				
				case 'name':
				
					if(!is_string($attributes[$key]) || empty($attributes[$key]))
						throw new \Exception("field '{$key}' must be a non empty string.");
					
					break;
				
				default:
					throw new \Exception("field '{$key}' unknown.");
			}
			
		}
		
		return true; 
	}
	
	
	protected function call(\Ething\Event\Signal $signal){
		$this->ething()->dispatchSignal( Ething\Event\Custom::emit($this->name) );
	}
	
}

File: ething/rule/action/EventTrigger.py
# coding: utf-8
from future.utils import string_types
from .. import Action
from ething.rule.event import Custom

class EventTrigger(Action):
    
    @staticmethod
    def validate (attributes, context):
        
        attributes.setdefault('name', None)
        
        for key in attributes:
            
            value = attributes[key]
            
            if key == 'name':
                
                if not( isinstance(value, string_types) and len(value)>0 ):
                    raise Exception("must be a non empty string.")
            
            else:
                raise Exception("%s: invalid" % key)
        
        return True
    
    
    
    def call(self, signal):
        self.ething.dispatchSignal(Custom.emit(self['name']))
    